src/bin/pgcopydb/ld_test_decoding.c

```diff
@@ -300,6 +302,65 @@ parseTestDecodingMessage(StreamContext *privateContext,
 }
 
 
+/*
+ * findIdentifierEndPos returns the first position of the separator in the given
+ * message after navigating through all the quotes. It assumes that the message
+ * contains a well-formed identifier and doesn't attempt to handle errors while
+ * parsing the message.
+ *
+ * This should always find an identifier end position, as we are parsing a
+ * well-formed message from the test_decoding plugin.
+ */
+static bool
+findIdentifierEndPos(const char *message, char separator, int *position)
+{
+	if (message == NULL || position == NULL)
+	{
+		return false;
+	}
+
+	int quoteCount = 0;
+
+	int pos = 0;
+
+	for (pos = 0; message[pos] != '\0'; pos++)
+	{
+		if (message[pos] == '"')
+		{
+			quoteCount++;
+		}
+
+		/*
+		 * We are looking for the first 'separator' in the message which
+		 * should be outside the quotes.
+		 *
+		 * When there is 'separator' inside quotes, quoteCount will be odd,
+		 * and we need to account it as a part of the identifier.
```

Comment:
I think that only works because of identifier escaping rules: it's using `""` to represent a `"` character, not `\"` like in other places, and also the separator itself can not be `"`. Can we add a check about the separator not being a double-quote character?